airflow/executors/mesos_executor.py

```diff
@@ -0,0 +1,422 @@
+
+            port_begin = 31000 + tid
+            port_end = 31000 + tid
+
+            self.log.debug("Launching task %d using offer %s", tid, offer['id']['value'])
+
+            task = {
+                'name': "AirflowTask %d" % tid,
+                'task_id': {'value': str(tid)},
+                'agent_id': {'value': offer['agent_id']['value']},
+                'resources': [
+                    {'name': 'cpus', 'type': 'SCALAR', 'scalar': {'value': self.task_cpu}},
+                    {'name': 'mem', 'type': 'SCALAR', 'scalar': {'value': self.task_mem}},
+                    {
+                        'name': 'ports',
+                        'type': 'RANGES',
+                        'ranges': {'range': [{'begin': port_begin, 'end': port_end}]},
+                    },
+                ],
+                'command': {
+                    'shell': 'true',
+                    'environment': {
+                        'variables': [
+                            {'name': 'AIRFLOW__CORE__SQL_ALCHEMY_CONN', 'value': self.core_sql_alchemy_conn},
+                            {'name': 'AIRFLOW__CORE__FERNET_KEY', 'value': self.core_fernet_key},
+                            {'name': 'AIRFLOW__CORE__LOGGING_LEVEL', 'value': 'DEBUG'},
+                        ]
+                    },
+                    'value': " ".join(cmd),
+                },
+                'container': {
+                    'type': 'DOCKER',
+                    'volumes': [
+                        {
+                            'container_path': self.mesos_docker_volume_dag_container_path,
+                            'mode': 'RO',
+                            'source': {
+                                'type': 'DOCKER_VOLUME',
+                                'docker_volume': {
+                                    'driver': self.mesos_docker_volume_driver,
+                                    'name': self.mesos_docker_volume_dag_name,
+                                },
+                            },
+                        },
+                        {
+                            'container_path': self.mesos_docker_volume_logs_container_path,
+                            'mode': 'RW',
+                            'source': {
+                                'type': 'DOCKER_VOLUME',
+                                'docker_volume': {
+                                    'driver': self.mesos_docker_volume_driver,
+                                    'name': self.mesos_docker_volume_logs_name,
+                                },
+                            },
+                        },
+                    ],
+                    'docker': {
+                        'image': self.mesos_slave_docker_image,
+                        'force_pull_image': 'true',
+                        'privileged': 'true',
+                        'parameters': [
+                            {'key': 'volume', 'value': self.mesos_docker_sock + ':/var/run/docker.sock'}
```

Comment:
Why do we need to mount the docker socket in the running slave image?

Reply:
That's something I hopefully can remove next week if I have enough tests in different environments. :-)

Reply:
FYI: The "deadline" for 2.0 is fast approaching (like 1-2 weeks) https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+2.0+-+Planning#Airflow2.0Planning-Nextsteps so this likely won't be included until 2.1 (which will be Feb or March most likely)

Reply:
I remove it. :-)

Reply:
ok, I reincluded it. It's necessary to have access to the docker.socks if u want to start a container inside of a container (thats the case with the DockerOperator). One more reason to write a MesosOperator.  :-)